addons/html_builder/static/src/website_sale/products_item_option_plugin.js

```diff
@@ -0,0 +1,430 @@
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { Plugin } from "@html_editor/plugin";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { rpc } from "@web/core/network/rpc";
+import { ProductsItemOption } from "./products_item_option";
+import { reactive } from "@odoo/owl";
+
+class ProductsItemOptionPlugin extends Plugin {
+  static id = "productsItemOptionPlugin";
+  static dependencies = ["history"];
+  itemSize = reactive({ x: 1, y: 1 });
+  count = reactive({ value: 0 });
+
+  resources = {
+    builder_options: [
+      {
+        OptionComponent: ProductsItemOption,
+        props: {
+          loadRibbons: this.loadRibbons.bind(this),
+          getDefaultSort: this.getDefaultSort.bind(this),
+          itemSize: this.itemSize,
+          count: this.count,
+        },
+        selector: "#products_grid .oe_product",
+        editableOnly: false,
+        title: _t("Product"),
+        groups: ["website.group_website_designer"],
+      },
+    ],
+
+    builder_actions: this.getActions(),
+  };
+
+  setup() {
+    this.currentWebsiteId = this.services.website.currentWebsiteId;
+    this.ribbonPositionClasses = {
+      left: "o_ribbon_left",
+      right: "o_ribbon_right",
+    };
+    this.loadRibbons().then((ribbons) => {
```

Comment:
this is actually annoying. this means that we perform the request for every pages, instead of only on pages with the current product grid. requests are expensive! would be better not to preload here, and only do it when the component ProductsItemOption is created.